dali/operators/image/resize/resize_attr.cc

```diff
@@ -0,0 +1,318 @@
+    const char *arg_name = has_resize_shorter_ ? "resize_shorter" : "resize_longer";
+    GetPerSampleArgument(res_x_, arg_name, spec, ws, N);
+
+    for (int i = 0; i < N; i++) {
+      for (int d = 0; d < spatial_ndim_; d++) {
+        in_size[d] = input_shape.tensor_shape_span(i)[d + first_spatial_dim_];
+        requested_size[d] = res_x_[i];
+      }
+
+      CalculateSampleParams(params_[i], requested_size, in_size);
+    }
+  } else if (has_size_) {
+    for (int i = 0; i < N; i++) {
+      for (int d = 0; d < spatial_ndim_; d++) {
+        in_size[d] = input_shape.tensor_shape_span(i)[d + first_spatial_dim_];
+        requested_size[d] = size_arg_.tensor_shape_span(i)[d];
+      }
+
+      CalculateSampleParams(params_[i], requested_size, in_size);
+    }
+  }
+}
+
+void ResizeAttr::AdjustOutputSize(float *out_size, const float *in_size) {
+  SmallVector<double, 3> scale;
+  SmallVector<bool, 3> mask;
+  scale.resize(spatial_ndim_, 1);
+  mask.resize(spatial_ndim_);
+
+  int sizes_provided = 0;
+  for (int d = 0; d < spatial_ndim_; d++) {
+    mask[d] = (out_size[d] != 0);
+    scale[d] = in_size[d] ? out_size[d] / in_size[d] : 1;
+    sizes_provided += mask[d];
+  }
+
+  if (sizes_provided == 0) {  // no clue how to resize - keep original size then
+    for (int d = 0; d < spatial_ndim_; d++)
+      out_size[d] = in_size[d];
+  } else {
+    if (mode_ == ResizeMode::Stretch) {
+      if (sizes_provided < spatial_ndim_) {
+        // if only some extents are provided - calculate average scale
+        // and use it for the missing dimensions;
+        double avg_scale = 1;
+        for (int d = 0; d < spatial_ndim_; d++) {
+          scale[d] = out_size[d] / in_size[d];
+          if (mask[d])
+            avg_scale *= std::abs(scale[d]);  // abs because of possible flipping
+        }
+        if (sizes_provided > 1)
+          avg_scale = std::pow(avg_scale, 1.0 / sizes_provided);
+        for (int d = 0; d < spatial_ndim_; d++) {
+          if (!mask[d]) {
+            scale[d] = avg_scale;
+            out_size[d] = in_size[d] * scale[d];
+          }
+        }
+      }
+      if (has_max_size_) {
+        for (int d = 0; d < spatial_ndim_; d++) {
+          if (max_size_[d] > 0 && std::abs(out_size[d]) > max_size_[d]) {
+            out_size[d] = std::copysignf(max_size_[d], out_size[d]);
+            scale[d] = out_size[d] / in_size[d];
+          }
+        }
+      }
+    } else {
+      // NotLarger or NotSmaller mode
+
+      // First, go through all the dimensions that have their scale defined and find
+      // the min (or max) scale
+      double final_scale = 0;
+      bool first = true;
+      for (int d = 0; d < spatial_ndim_; d++) {
+        if (mask[d]) {
+          float s = std::abs(scale[d]);
+          if (first ||
+              (mode_ == ResizeMode::NotSmaller && s > final_scale) ||
+              (mode_ == ResizeMode::NotLarger && s < final_scale))
+            final_scale = s;
+          first = false;
```

Comment:
I think you can add break here.

Reply:
No, this loop must go to the end. The variable `first` is here to put whatever the first defined scale in `final_scale` and then just update it when it matches the predicate for current resize mode.